pkg/flow/flow_mem_driver.go

```diff
@@ -805,6 +805,15 @@ func (fc *FlowCollector) updateRecord(record interface{}) error {
 		if link, ok := record.(LinkRecord); ok {
 			if current, ok := fc.Links[link.Identity]; !ok {
 				if link.StartTime > 0 && link.EndTime == 0 {
+					// adding a new link, can only have one link between routers
```

Comment:
I would suggest removing the "peer" link record when the local link record is deleted. For example, a link for the parent may not get created if there is an overall van topology change. Also, the link name may be the same as the deleted link.

Reply:
+1 to this suggestion!

I am not certain there will be anything useful to reuse here, but I've somewhat recently added similar logic [here](https://github.com/skupperproject/skupper/blob/72e6df22bc8bc764fd4c4574d56bfe48677fb345/pkg/flow/flow_mem_driver.go#L2798-L2904) for counting links in some of our operational metrics.